wgpu-types/src/lib.rs

```diff
@@ -1477,6 +1841,95 @@ impl Default for Extent3d {
     }
 }
 
+impl Extent3d {
+    /// Calculates the [physical size] is backing an texture of the given format and extent.
+    /// This includes padding to the block width and height of the format.
+    ///
+    /// This is the size that you must upload at when uploading to _mipmaps_ of compressed textures.
```

Comment:
worth noting that you can only use this size for `write_texture`. If you are using `copy_buffer_to_texture`, you'd have to also respect the bytes-per-row alignment.

Reply:
Clarified that this was only for the texture extent.